packages/protocol/contracts/shared/common/EssentialContractStorage.sol

```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+/// @title EssentialContractStorage
+/// @notice Mirrors the storage footprint of `EssentialContract` so that contracts
+///         replacing it can keep the exact slot layout without inheriting its logic.
+contract EssentialContractStorage {
```

Comment:
Since this contract is only used by Anchor.sol, so maybe make sense to move it to Anchor.sol?